src/test/java/uk/gov/hmcts/dts/fact/services/admin/list/AdminAreasOfLawServiceTest.java

```diff
@@ -44,4 +46,120 @@ void shouldReturnAllAreasOfLaw() {
+    }
+
+    @Test
+    void shouldUpdateAreaOfLaw() {
+        final uk.gov.hmcts.dts.fact.model.admin.AreaOfLaw areaOfLaw =
+            new uk.gov.hmcts.dts.fact.model.admin.AreaOfLaw(getTestAreaOfLawEntities().get(0));
+        areaOfLaw.setExternalLink("https://something.else.link");
+        areaOfLaw.setExternalLinkDescription("A different description");
+        areaOfLaw.setExternalLinkDescriptionCy("A different description in Welsh");
+
+        when(areasOfLawRepository.findById(areaOfLaw.getId()))
+            .thenReturn(Optional.of(getTestAreaOfLawEntities().get(0)));
+        when(areasOfLawRepository.save(any(AreaOfLaw.class)))
+            .thenAnswer((Answer<AreaOfLaw>) invocation -> invocation.getArgument(0));
+
+        assertThat(areasOfLawService.updateAreaOfLaw(areaOfLaw)).isEqualTo(areaOfLaw);
+    }
+
+    @Test
+    void updateShouldThrowNotFoundExceptionWhenAreaOfLawDoesNotExist() {
+        final AreaOfLaw testAreaOfLaw = getTestAreaOfLawEntities().get(0);
+        when(areasOfLawRepository.findById(testAreaOfLaw.getId())).thenReturn(Optional.empty());
+
+        assertThatThrownBy(() -> areasOfLawService
+            .updateAreaOfLaw(new uk.gov.hmcts.dts.fact.model.admin.AreaOfLaw(testAreaOfLaw)))
+            .isInstanceOf(NotFoundException.class);
+
+        verify(areasOfLawRepository, never()).save(any());
+    }
+
+    @Test
+    void shouldCreateAreaOfLaw() {
+        final uk.gov.hmcts.dts.fact.model.admin.AreaOfLaw areaOfLaw =
+            new uk.gov.hmcts.dts.fact.model.admin.AreaOfLaw();
+        areaOfLaw.setName("New Area of Law");
+        areaOfLaw.setExternalLink("https://newarea.of.law");
+        areaOfLaw.setExternalLinkDescription("This is a new area of law");
+        areaOfLaw.setExternalLinkDescriptionCy("This is a new area of law - welsh");
+        areaOfLaw.setDisplayName("This is new area of law display name");
+        areaOfLaw.setDisplayNameCy("This is new area of law display name - welsh");
+        areaOfLaw.setDisplayExternalLink("https://external.newarea.of.law");
+
+        when(areasOfLawRepository.save(any(AreaOfLaw.class))).thenAnswer((Answer<AreaOfLaw>)invocation -> invocation.getArgument(0));
+
+        assertThat(areasOfLawService.createAreaOfLaw(areaOfLaw)).isEqualTo(areaOfLaw);
+    }
+
+    @SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+    @Test
+    void createShouldThrowDuplicatedListItemExceptionIfAreaOfLawAlreadyExists() {
+        final List<AreaOfLaw> mockAreasOfLaw = getTestAreaOfLawEntities();
+        when(areasOfLawRepository.findAll()).thenReturn(mockAreasOfLaw);
+
+        final uk.gov.hmcts.dts.fact.model.admin.AreaOfLaw areaOfLaw =
+            new uk.gov.hmcts.dts.fact.model.admin.AreaOfLaw(getTestAreaOfLawEntities().get(0));
+
+        assertThatThrownBy(() -> areasOfLawService
+            .createAreaOfLaw(areaOfLaw))
+            .isInstanceOf(DuplicatedListItemException.class);
+    }
+
+    private List<AreaOfLaw> getTestAreaOfLawEntities() {
```

Comment:
This is the same issue as in the controller test class. A new array is recreated every time the method is called. Would be better to set a constant arrays at the class level. If you don't want the top of the class to look to clustered. You can just do the following but make the method static:

private static final List<AreaOfLaw> AREAS_OF_LAW = getTestAreaOfLawEntities();